src/Analyzers/CSharp/Tests/RemoveUnnecessarySuppressions/RemoveUnnecessarySuppressionsTests.cs

```diff
@@ -0,0 +1,186 @@
+
+namespace N
+{{
+    class C
+    {{
+        public int F;
+        public int P {{ get; }}
+        public void M() {{ }}
+    }}
+}}";
+            await VerifyCS.VerifyCodeFixAsync(input, input);
+        }
+
+        [Theory]
+        // Field - no matching symbol
+        [InlineData(@"Scope = ""member""", @"Target = ""~F:N.C.F2""", "assembly")]
+        // Field - no matching symbol (case insensitive)
+        [InlineData(@"Scope = ""Member""", @"Target = ""~F:N.C.F2""", "assembly")]
+        [InlineData(@"Scope = ""MEMBER""", @"Target = ""~F:N.C.F2""", "assembly")]
+        // Property - invalid scope
+        [InlineData(@"Scope = ""invalid""", @"Target = ""~P:N.C.P""", "assembly")]
+        // Method - wrong signature
+        [InlineData(@"Scope = ""member""", @"Target = ""~M:N.C.M(System.Int32)""", "assembly")]
+        // Method - module scope
+        [InlineData(@"Scope = ""module""", @"Target = ""~M:N.C.M()""", "assembly")]
+        // Method - null scope
+        [InlineData(@"Scope = null", @"Target = ""~M:N.C.M()""", "assembly")]
+        // Method - no scope
+        [InlineData(null, @"Target = ""~M:N.C.M()""", "assembly")]
+        // Member scope - null target
+        [InlineData(@"Scope = ""member""", @"Target = null", "assembly")]
+        // Member scope - no target
+        [InlineData(@"Scope = ""member""", null, "assembly")]
+        // Type - no matching namespace
+        [InlineData(@"Scope = ""type""", @"Target = ""~T:N2.C""", "assembly")]
+        // Namespace - extra namespace qualification
+        [InlineData(@"Scope = ""namespace""", @"Target = ""~N:N.N2""", "assembly")]
+        // NamespaceAndDescendants - empty target
+        [InlineData(@"Scope = ""namespaceanddescendants""", @"Target = """"", "assembly")]
+        // Module - no scope, empty target
+        [InlineData(null, @"Target = """"", "assembly")]
+        // Module - no scope, non-empty target
+        [InlineData(null, @"Target = ""~T:N.C""", "assembly")]
+        // Module scope, empty target
+        [InlineData(@"Scope = ""module""", @"Target = """"", "assembly")]
+        // Module no scope, non-empty target
+        [InlineData(@"Scope = ""module""", @"Target = ""~T:N.C""", "assembly")]
+        public async Task InvalidSuppressions(string? scope, string? target, string attributeTarget)
+        {
+            var scopeString = scope != null ? $@", {scope}" : string.Empty;
+            var targetString = target != null ? $@", {target}" : string.Empty;
+
+            var input = $@"
+[{attributeTarget}: [|System.Diagnostics.CodeAnalysis.SuppressMessage(""Category"", ""Id: Title"", Justification = ""Pending""{scopeString}{targetString})|]]
+
+namespace N
+{{
+    class C
+    {{
+        public int F;
+        public int P {{ get; }}
+        public void M() {{ }}
+    }}
+}}";
+
+            var fixedCode = $@"
+
```

Comment:
💭 It would be nice if this second blank line was removed (applies for the case where the attribute is preceded by a blank line).

Reply:
Sure, I can file an issue for the syntax generator to handle this case for RemoveNode(s) API.

Reply:
Filed https://github.com/dotnet/roslyn/issues/44342